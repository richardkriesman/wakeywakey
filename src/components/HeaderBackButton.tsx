/**
 * @module components
 */

import * as React from "react";
import { ReactNode } from "react";
import { StyleSheet } from "react-native";
import { Button } from "react-native-elements";

import Colors from "../constants/Colors";

export interface HeaderBackButtonProps {
    title: string;
    onPress: () => void;
}

export class HeaderBackButton extends React.Component<HeaderBackButtonProps> {
    public constructor(props: HeaderBackButtonProps) {
        super(props);
    }

    public render(): ReactNode {
        return (
            <Button type="clear" titleStyle={styles.title} title={this.props.title}
                    onPress={this.props.onPress}/>
        );
    }
}

const styles = StyleSheet.create({
    title: {
        color: Colors.appleButtonRed,
        marginLeft: 10
    }
});
