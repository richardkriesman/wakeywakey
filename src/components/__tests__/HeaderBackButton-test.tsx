import * as React from "react";
import renderer from "react-test-renderer";
import { HeaderBackButton } from "../HeaderBackButton";

const createTestProps = (props: object) => ({
    navigation: {},
    onPress: jest.fn(),
    ...props
});

describe("HeaderBackButton", () => {

    let props: any;

    beforeEach(() => {
        props = createTestProps({ title: "Test Button" });
    });

    it("renders correctly", () => {
        const tree = renderer.create(<HeaderBackButton {...props} />).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it("calls the correct navigation function onPress", () => {
        const button = renderer.create(<HeaderBackButton {...props} />);
        button.root.findByType(HeaderBackButton).props.onPress();
        expect(props.onPress).toHaveBeenCalledTimes(1);
    });

});
