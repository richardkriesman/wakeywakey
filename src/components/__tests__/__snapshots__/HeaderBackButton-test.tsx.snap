// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeaderBackButton renders correctly 1`] = `
<View
  style={
    Object {
      "borderRadius": 3,
    }
  }
>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "transparent",
          "borderColor": "#2089dc",
          "borderRadius": 3,
          "borderWidth": 0,
          "flexDirection": "row",
          "justifyContent": "center",
          "padding": 8,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#ff3b30",
            "fontSize": 18,
            "marginLeft": 10,
            "paddingBottom": 1,
            "paddingTop": 2,
            "textAlign": "center",
          }
        }
      >
        Test Button
      </Text>
    </View>
  </View>
</View>
`;
